### ISSUE TITLE
how can i check videoSrc is right(when src is wrong, the videocontext is stop, can get me some prompt)

### ISSUE THREAD

**Issue by Statfine** [bbc/VideoContext#38]
this.ctx = new VideoContext();
var videoNode = this.ctx.createVideoSourceNode('**errorvideo.mp4**');
videoNode.start(0);
videoNode.stop(10);
var videoNode2 = ctx.createVideoSourceNode('errorvideo.mp4');
videoNode2.start(10);
videoNode2.stop(20);

the first videoNode will still loading, so how can i check the src is right; or can you give me a prompt when it is wrong like 'registerCallback('nocontent', ... )'

**Comment by MatthewShotton** [comment]
Hi @Statfine 

Thanks for raising this. I've just pushed a new release, v0.36.0 which adds an error callback to nodes which will get thrown if content cannot be loaded for a specific node. It can be used like the following:

```JavaScript
this.ctx = new VideoContext();
let video = ctx.video("./errorvideo.mp4");
video.start(0);

video.registerCallback("error", (n)=>{
    console.log("Error with node", n);
   // Destroy the node so it doesn't try and play and cause the videocontext to stall.
    n.destroy();
});
```

Currently you have to call ```destroy()``` or ```resetTimelineState()``` on a node which throws an error otherwise the VideoContext will continue trying to play it which will cause the VideoContext to halt.

Does that help with your requirements?

Kind regards,
Matt

**Comment by Statfine** [comment]
@MatthewShotton  can videoContext still play if i don't do n.destroy() when video.registerCallback("error",...); just don't show frame

**Comment by MatthewShotton** [comment]
@Statfine I've just pushed a change to v0.38.0 which will let the VideoContext continue playing if one of the source nodes throws an error. Let me know if you have any problems with it. Cheers!

**Comment by Statfine** [comment]
![wechatimg43](https://cloud.githubusercontent.com/assets/4608910/24945808/5be84792-1f93-11e7-97f9-4b9455505240.jpeg)

i have update it to v0.38.0; but the VideoContext is not playing when source nodes throws an error, should i use something function to continue

**Comment by MatthewShotton** [comment]
hi @Statfine , I've pushed v0.38.1 which adds some extra checks for testing if a video node is not valid. 

If that doesn't fix the problem are you able to put together a minimal example that demonstrates the problem you're seeing? I'm having trouble re-creating the exact error you have.

**Comment by Statfine** [comment]
const urlOne = 'rightUrl';
const urlTwo = 'wrongUrl'
const node = this.ctx.video(urlOne, 0, 4, { volume: 0.8, loop: false });
node.startAt(0);
node.stopAt(15);

const nodeTwo = this.ctx.video(urlTwo, 0, 4, { volume: 0.8, loop: false });
nodeTwo.startAt(15);
nodeTwo.stopAt(30);

node.connect(this.ctx.destination);
nodeTwo.connect(this.ctx.destination);
this.ctx.play();
![wechatimg53](https://cloud.githubusercontent.com/assets/4608910/24951111/e94c25ec-1fa5-11e7-8e8a-b46feae4ccfc.jpeg)

**Comment by MatthewShotton** [comment]
@Statfine I've just published v0.38.2 to npm/github. Let me know if you have any luck with that! 

I'm not sure why my original test code failed to catch this, but the code you've sent has replicated the issue well enough to hopefully fix it.

**Comment by Statfine** [comment]
thanks, VideoContext is still playing when source nodes throws an error after i update it to v0.38.2.It's very helpful for me
